apps/voting/app/src/components/VoteActions.js

```diff
@@ -39,16 +38,24 @@ const VoteActions = React.memo(({ vote, onVoteYes, onVoteNo, onExecute }) => {
   const hasVoted = [VOTE_YEA, VOTE_NAY].includes(connectedAccountVote)
 
   useEffect(() => {
+    let cancelled = false
```

Comment:
Hmm, for this you could've used the `usePromise` hook that the inner functions rely on.

Reply:
Hmm yes, but `usePromise()`'s API is based on returning the resulting value from the promise (so you can use it) whereas we just want to cancel it here and avoid a side effect.

Reply:
Pseudo code here (this is what I was thinking):

``` JavaScript
const promise = Promise.all([canUserVotePromise, canExecutePromise, userBalancePromise])
const value = usePromise(promise, false)
setReady(value)
```

But yeah, there are other things that might need to be worked on.

Reply:
Yeah, I'm worried there might be funny things with how the state gets updated / triggered (compared to now, which is relatively obvious what happens on cancellation / updates), and to be honest `usePromise()` is something we should find a better abstraction for since it's kind of clunky.